### PR TITLE
Add documentation for template vacuum component

### PR DESCRIPTION
**Pull request by luukdobber**
**Description:**
Adds documentation for the Template Vacuum component.

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#17215

## Checklist:

- [x] Branch: `next` is for changes and new documentation that will go public with the next [home-assistant](https://github.com/home-assistant/home-assistant) release. Fixes, changes and adjustments for the current release should be created against `current`.
- [x] The documentation follow the [standards][standards].

[standards]: https://home-assistant.io/developers/documentation/standards/

### CODE REVIEW

File: source/_components/vacuum.template.markdown [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+```yaml
+# Example configuration.yaml entry
+vacuum:
+  - platform: template
+    vacuums:
+      living_room_vacuum:
+        friendly_name: "Living room vacuum"
+        value_template: "{{ states('input_select.vacuum_state') }}"
+        battery_level_template: "{{ states('input_number.vacuum_battery_level') | int }}"
+        fan_speed_template: "{{ states('input_select.vacuum_fan_speed') }}"
```
Comment (WofWca) [inline source/_components/vacuum.template.markdown]: The standard says that there must be a sample configuration with only required options.

Comment (WofWca) [reply source/_components/vacuum.template.markdown]: But you can put additional examples below.

**Comment by WofWca** [comment]
But you can put additional examples below.

File: source/_components/vacuum.template.markdown [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+---
+
+The `template` platform creates vacuums that combine components and provides the
+ability to run scripts or invoke services for each of the start, pause, stop,
+return_to_base, clean_spot, locate and set_fan_speed commands of a vacuum.
+
+To enable Template Vacuums in your installation, add the following to your
+`configuration.yaml` file:
+
+{% raw %}
```
Comment (frenck) [inline source/_components/vacuum.template.markdown]: There is no need to raw this.

**Review by frenck — CHANGES_REQUESTED** [review]
Thanks for the PR, @luukd.

I've left you a tiny comment, and just like @WofWca mentioned already.
It is definitely cool to add an extra example, but it should go below the configuration list.

Thanks already 👍

File: source/_components/vacuum.template.markdown (hunk 2 above)
Comment (luukdobber) [reply source/_components/vacuum.template.markdown]: Fixed. I've also added an example that combines the Template Vacuum with the Harmony Hub and an example containing Jinja2 templates.

**Review by frenck — APPROVED** [review]

**Comment by frenck** [comment]
Thanks for the updates, @luukd! 👍 

✅ Approved. Can be merged as soon as the parent PR gets merged.
⚠️ The `ha_release` might need an update before merging this one in.

**Comment by luukdobber** [comment]
> This issue has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs. Thank you for your contributions.

I have resolved the last issues in the parent PR. Can this PR stay open while the parent PR awaits review?

**Comment by frenck** [comment]
Of course 👍

**Comment by frenck** [comment]
Closing PR, since the parent PR has been closed.

**Pull request closed without merging**